lib/unread/scopes.rb

```diff
@@ -11,11 +11,24 @@ def join_read_marks(user)
       end
 
       def unread_by(user)
-        result = join_read_marks(user).
-                 where('read_marks.id IS NULL')
+        result = join_read_marks(user)
 
         if global_time_stamp = user.read_mark_global(self).try(:timestamp)
-          result = result.where("#{table_name}.#{readable_options[:on]} > ?", global_time_stamp)
+          result = result.where("read_marks.id IS NULL AND #{table_name}.#{readable_options[:on]} > ?", global_time_stamp)
+        else
+          result = result.where('read_marks.id IS NULL')
+        end
+
+        result
+      end
+
+      def read_by(user)
+        result = join_read_marks(user)
+
+        if global_time_stamp = user.read_mark_global(self).try(:timestamp)
+          result = result.where("read_marks.id IS NOT NULL OR #{table_name}.#{readable_options[:on]} <= ?", global_time_stamp).uniq
```

Comment:
Hm, the LEFT JOIN condition makes sure there is only 0 or 1 record joined, so duplicates are impossible. Because using DISTINCT can slow down queries a lot, I would prefer to remove the `uniq`.